components/datetime/src/options/length.rs

```diff
@@ -98,6 +99,27 @@ impl Default for Bag {
     }
 }
 
+impl Bag {
+    /// Constructs a Bag with all fields set to None
+    ///
+    /// Note that the [`Default`] implementation returns long date and time options
+    pub fn empty() -> Self {
+        Self {
+            date: None,
+            time: None,
+            preferences: None,
+        }
+    }
+
+    /// Constructs a Bag given a date and time field (preferences set to None)
+    pub fn from_date_time_style(date: Option<Date>, time: Option<Time>) -> Self {
```

Comment:
Thought: This appears to be just a style change, and not technically required by this patch. You can still construct a non-exhaustive struct using the old style of construction. However, it seems like this way encourages code that won't break when new fields are added to structs.

Reply:
You cannot use the old style of construction anymore. See https://github.com/rust-lang/lang-team/issues/143

I've added this method to make it pleasant to construct